pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -168,7 +168,8 @@ func (c *Cluster) makeDeployment(nodeName string, devices []rookalpha.Device, se
 		configVolumeMounts = append(configVolumeMounts, binariesMount)
 	} else {
 		// other osds can launch the osd daemon directly
-		command = []string{"ceph-osd"}
+		command = append([]string{"ceph-osd",
+			fmt.Sprintf("--public-addr=$(%s)", k8sutil.PublicIPEnvVar)})
```

Comment:
Why do we need this command line arg? The public IP env var is already passed in with [this var](https://github.com/rook/rook/blob/master/pkg/operator/ceph/cluster/osd/spec.go#L117), which could then be superseded by the override yaml. Perhaps you needed this here so that the override yaml wouldn't mess with this parameter since the parameter takes precedence. If you remove the public addr from the config override, you may not need this parameter.

Reply:
At some point after submitting this PR I believe that `pkg/operator/ceph/cluster/osd/pod.go` was refactored/renamed to `pkg/operator/ceph/cluster/osd.osd.go`.  When refactoring this change to work with master, there was no longer a `tini` being passed in and this change you see here is a result of that.  As far as I can remember, this latest batch of changes was indeed built and tested.

Reply:
yeah, there were certainly a number of changes around this, so I don't doubt it was working and could have been broken your PR. you're saying you already tested since those changes were merged?